io.c

```diff
@@ -13331,6 +13331,163 @@ set_LAST_READ_LINE(VALUE val, ID _x, VALUE *_y)
+ *  - #reopen:: Reassociates +self+ with a new or existing I/O stream.
+ *  - #rewind:: Positions +self+ to the beginning of input.
+ *  - #seek:: Sets the offset for +self+.
+ *  - #tell:: Returns the current offset (bytes).
+ *
+ *  === Iterating
+ *
+ *  - ::foreach:: Calls the given block with each successive file line
+ *                for the given filepath.
+ *  - #each (aliased as #each_line):: Calls the given block
+ *                                    with each successive line in +self+.
+ *  - #each_byte:: Calls the given block with each successive byte in +self+.
+ *  - #each_char:: Calls the given block with each successive character in +self+.
+ *  - #each_codepoint:: Calls the given block with each successive codepoint in +self+.
+ *  - #each_line:: Calls the given block with each successive codepoint in +self+.
+ *
+ *  === Converting
+ *
+ *  - ::try_convert:: Returns the new \IO object resulting from converting
+ *                    the given object.
+ *  - #inspect:: Returns the string representation of +self+.
+ *
+ *  === Settings
+ *
+ *  - #autoclose=:: Sets whether +self+ auto-closes.
+ *  - #binmode:: Sets +self+ to binary mode.
+ *  - #close_on_exec=:: Sets the close-on-exec flag.
+ *  - #sync=:: Sets the sync-mode to the given value.
+ *
+ *  === Querying
+ *
+ *  - #autoclose?:: Returns whether +self+ auto-closes.
+ *  - #binmode?:: Returns whether +self+ is in binary mode.
+ *  - #close_on_exec?:: Returns the close-on-exec flag for +self+.
+ *  - #closed?:: Returns whether +self+ is closed.
+ *  - #eof? (aliased as #eof):: Returns whether +self+ is at end-of-file.
+ *  - #fileno (aliased as #to_i):: Returns the integer file descriptor for +self+
+ *  - #sync:: Returns whether +self+ is in sync-mode.
+ *  - #tty (aliased as #isatty):: Returns whether +self+ is a terminal.
+ *
+ *  === Closing
+ *
+ *  - #close:: Closes +self+.
+ *  - #close_read:: Closes +self+ for reading.
+ *  - #close_write:: Closes +self+ for writing.
+ *
+ *  === Encoding
+ *
+ *  - #external_encoding:: Returns the external encoding object for +self+.
+ *  - #internal_encoding:: Returns the internal encoding object for +self+.
+ *  - #set_encoding:: Sets the encoding for +self+.
+ *  - #set_encoding_by_bom:: Sets the encoding for +self+, based on its
+ *                           Unicode byte-order-mark.
+ *
+ *  === Buffering
+ *
+ *  - #fdatasync:: Immediately writes all buffered data in +self+ to disk.
+ *  - #flush:: Flushes any buffered data within +self+ to the underlying
+ *             operating system.
+ *  - #fsync:: Immediately writes all buffered data in +self+ to disk.
+ *  - #ungetbyte:: Pushes the given byte or string onto +self+.
+ *  - #ungetc:: Pushes the given byte or string onto +self+.
+ *
+ *  === Low-Level Access
+ *
+ *  - ::sysopen:: Opens the file given by its path,
+ *                returning the integer file descriptor.
+ *  - #fcntl:: Passes a low-level command to the file specified
```

Comment:
`See operating system documentation for fcntl system call.`  This is assuming we want to document this at all.  Nobody who doesn't already know what the fcntl system call does would want to use this.  Same for `ioctl`.

Reply:
Doesn't this belong at the method's doc, below?  "What's Here" is just a summary?

Reply:
It's impossible to have a simple summary of these methods.  As the method documentation states, arguments and results for both are platform dependent.  We should either tell people to read the man page, or not document the methods here at all, IMO.